### PR TITLE
refactor: to update tooling (rollup, babel et al) to recent versions

### PR DESCRIPTION
**Pull request by tdeekens**
> This pull request updates all dependencies related to tooling to their recent versions.

### Background & Context

The versions were quite outdated. Before we proceed to take on CI and maybe other things I suggest we get our tooling stack up to date. This includes:

1. Babel (it's all scoped packages by now)
2. Rollup (moved quite quickly and we picked the right bundler two years back)
3. Prettier chores the code a bit but that's the nature of formatters
4. ESLint and XO changes a bit. I disabled some rules I didn't want to use in code. Some rather okay suggestions I changed
5. Suggesting to use terser instead of uglify to minify our code

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+13
```
Comment (tdeekens) [inline .nvmrc:1]: Just helps when working locally that we all use the same node version. Plus we can use it later on CI.

File: rollup.config.js [hunk 2]
```diff
@@ -1,44 +1,37 @@
-const nodeResolve = require("rollup-plugin-node-resolve");
-const babel = require("rollup-plugin-babel");
-const replace = require("rollup-plugin-replace");
-const uglify = require("rollup-plugin-uglify");
-const fs = require("fs");
+const nodeResolve = require('rollup-plugin-node-resolve');
+const babel = require('rollup-plugin-babel');
+const replace = require('rollup-plugin-replace');
+const { terser } = require('rollup-plugin-terser');
```
Comment (tdeekens) [inline rollup.config.js:4]: I know this is a sensitive topic here and I am not forcing this. I just wanted to bring up the suggestion. Terser is quite happily adopted in the community by now. The main reason is that uglify didn't move to support newer language features if I recall.

1. https://github.com/terser/terser
2. https://github.com/mishoo/UglifyJS2

I am fine either way but didn't have any bad experiences with terser.

File: src/purify.js [hunk 3]
```diff
@@ -947,7 +949,7 @@ function createDOMPurify(window = getGlobal()) {
         }
 
         arrayPop(DOMPurify.removed);
-      } catch (error) {}
+      } catch (_) {}
```
Comment (tdeekens) [inline src/purify.js:952]: This is ESLint fixes. Cause we don't use the arg to indicate that better.

File: test/karma.conf.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-const commonjs = require("rollup-plugin-commonjs");
+const commonjs require('@rollup/plugin-commonjs');
```
Comment (tdeekens) [inline test/karma.conf.js:1]: The other plugin is deprecated.

**Comment by tdeekens** [comment]
Empty TravisCI logs. What’s up there. Lets see another day.

**Comment by tdeekens** [comment]
The main problem here is that Node.js moved long on and is on v13 now [ref](https://nodejs.org/en/about/releases/).

CleanShot 2020-04-19 at 10 01 49@2x

That doesn't mean we can't support these older versions but unlikely can test on them cause all dependencies have moved on. Meaning fsevents or cross-env doesn't support the old versions anymore.

Maybe we can note something in the readme at a later state which versions we actively support and which ones we can't guarantee support for?

**Comment by tdeekens** [comment]
Yay, at least now it's a test failing. Maybe @cure53 you have an idea. That's something again I know not too much about. Do you see any change here that could cause this?

File: rollup.config.js (hunk 2 above)
Comment (cure53) [reply rollup.config.js:4]: :+1: Fine with trying terser

File: src/purify.js (hunk 3 above)
Comment (cure53) [reply src/purify.js:952]: Yeah, had to do the same a while ago

**Comment by cure53** [comment]
> The main problem here is that Node.js moved long on and is on v13 now [ref](https://nodejs.org/en/about/releases/).
> 
> CleanShot 2020-04-19 at 10 01 49@2x
> 
> That doesn't mean we can't support these older versions but unlikely can test on them cause all dependencies have moved on. Meaning fsevents or cross-env doesn't support the old versions anymore.
> 
> Maybe we can note something in the readme at a later state which versions we actively support and which ones we can't guarantee support for?

I think, just as with older browsers, we need to phase out support for older versions. Happy to draft a text for the README once we have clarity with versions will be supported.

**Comment by tdeekens** [comment]
My goal would be to get GitHub Actions in. Then rebase this ontop master and then move on with this.

**Comment by tdeekens** [comment]
> I think, just as with older browsers, we need to phase out support for older versions. Happy to draft a text for the README once we have clarity with versions will be supported.

Yeah I agree. I think bottom line it's that we don't break with older Node.js versions yet and they continue to work. We however don't test for them anymore. Implying that potentially regressions will not be spotted from now on (if we merge these updates here). In future terms it would also mean that we similarily follow the Node.js release cycle and support any active version. However, phase out support for any version entering maintainance. As we do now with this.

**Comment by cure53** [comment]
> Yeah I agree. I think bottom line it's that we don't break with older Node.js versions yet and they continue to work. We however don't test for them anymore.

This makes perfect sense!

**Comment by cure53** [comment]
Is this one ready for merging as well?

**Comment by tdeekens** [comment]
Bloody hell. This is more fun than I thought. Seems like no dependency moved for 3 years. I reduced it to a meaningful update. From here we can move stepwise if possible and worthwhile for the lib :)

**Comment by tdeekens** [comment]
Yes, I will merge this once it passes.

File: README.md [hunk 5]
```diff
@@ -40,11 +40,11 @@ Afterwards you can sanitize strings by executing the following code:
 var clean = DOMPurify.sanitize(dirty);
 ```
 
-The resulting HTML can be written into a DOM element using `innerHTML` or the DOM using `document.write()`. That is fully up to you. But keep in mind, if you use the sanitized HTML with jQuery's very insecure `elm.html()` method, then the `SAFE_FOR_JQUERY` flag has to be set to make sure it's safe! Other than that, all is fine. 
+The resulting HTML can be written into a DOM element using `innerHTML` or the DOM using `document.write()`. That is fully up to you. But keep in mind, if you use the sanitized HTML with jQuery's very insecure `elm.html()` method, then the `SAFE_FOR_JQUERY` flag has to be set to make sure it's safe! Other than that, all is fine.
```
Comment (tdeekens) [inline README.md:43]: Just auto-formatting. Will follow-up with an npm run script for it.

File: test/test-suite.js [hunk 6]
```diff
@@ -1,668 +1,1453 @@
-module.exports = function(DOMPurify, window, tests, xssTests) {
+module.exports = function (DOMPurify, window, tests, xssTests) {
```
Comment (tdeekens) [inline test/test-suite.js:1]: Oh sorry. Didn't intend to change this. Will revert it to master.

**Pull request merged**